framework/src/auxkernels/TagVectorArrayAux.C

```diff
@@ -0,0 +1,47 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TagVectorArrayAux.h"
+
+registerMooseObject("MooseApp", TagVectorArrayAux);
+
+InputParameters
+TagVectorArrayAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+
+  params.addRequiredParam<TagName>("vector_tag", "Tag Name this Aux works on");
+  params.addRequiredCoupledVar("v",
+                               "The coupled variable whose components are coupled to AuxVariable");
+  params.set<ExecFlagEnum>("execute_on", true) = {EXEC_TIMESTEP_END};
+  params.addRequiredParam<unsigned int>(
+      "component",
+      "What component to index the variable at. This is a relevant "
+      "parameter in the context of array variables.");
+
+  params.addClassDescription("Couple a tag vector, and return its nodal value");
+  return params;
+}
+
+TagVectorArrayAux::TagVectorArrayAux(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _v(coupledVectorTagArrayValue("v", "vector_tag")),
+    _component(getParam<unsigned int>("component"))
+{
+  if (getArrayVar("v", 0)->feType() != _var.feType())
+    paramError("variable",
+               "The AuxVariable this AuxKernel is acting on has to have the same order and family "
+               "as the variable 'v'");
+}
+
+Real
+TagVectorArrayAux::computeValue()
```

Comment:
Should override `compute()` like `BuildArrayVariableAux` to directly manipulate dofs.

Reply:
I should not have to do that, right? I should be able to still just override `computeValue` like `FunctionArrayAux` does, right?

Reply:
It is more costly. Implementing `compute()` is quite simple.

Reply:
It is strange when you bring up cost. When I am reviewing your code, you are not worried about cost, and then when you are reviewing mine, you are worried about cost. :laughing: Have you profiled this? Are the 9x more lines giving you much speed up? It is sure nice to just return `_v[_qp]` in a single-line implementation

Reply:
I typically care simpler codes but I care more whether we are doing things in a more direct way, i.e. if data are held in the right places, if we activates unnecessary code pathes, etc. Here manipulating dofs is the direct way. A by-product is the performance because it by pass the local mass matrix inversion (I do not care this much).

Reply:
I just get really tired of your seeming double standards. You always demand that we review your code quickly and you always push back when we suggest changes. "Unnecessary code paths". You mean like writing a one-to-all algorithm in a few lines of code as opposed to an all-to-all in a single line?

Reply:
I will make your suggested change

Reply:
Actually it seems to me like your suggested change will not work unless the array variable is Lagrange. We would need to make a tag version of `coupledAllDofValues` in order for your change to make sense

Reply:
Yeah, we always push back. I understand. I have leant to listen reviewers ;-)

Reply:
Wait, I am checking.

Reply:
That is true, we only have `coupledArrayDofValues` but not `coupledVectorTagArrayDofValues` yet. Maybe add that? Sorry for this.

Reply:
I will probably delete `coupledVectorTagArrayValue` then, since there will still be no use for it